pkg/checker.go

```diff
@@ -0,0 +1,170 @@
+package pkg
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+const (
+	SuccessColor = "\033[1;32m%s\033[0m"
+	ErrorColor   = "\033[1;31m%s\033[0m"
+	DebugColor   = "\033[0;36m%s\033[0m"
+)
+
+/*type URLStatus struct {
+	url    string
+	err    string
+	status int
+	index  int
+}*/
+
+type URLStatus struct {
+
+	// err
+	Err string `json:"err,omitempty"`
+
+	// index
+	Index int64 `json:"index,omitempty"`
+
+	// status
+	Status int64 `json:"status,omitempty"`
+
+	// url
+	URL string `json:"url,omitempty"`
+}
+
+func checkWebsite(website string) (string, error) {
+	var err error
+	websiteTries := [3]string{website, "https://" + website, "http://" + website}
+	for _, web := range websiteTries {
+		_, err = http.Get(web)
+		if err == nil {
+			website = web
+		}
+	}
+	return website, err
+}
+
+// get links inside a given website
+func GetLinks(website string) ([]string, error) {
```

Comment:
Split into these methods

- `GetPageSource(website string) string` 
-  `ExtractLinksFromContent(pageSource string) []string`
This will make it testable

Reply:
optionally you can make io.Reader if you want to instead of string